### PR TITLE
Feature/infinite-scroll

### PR DESCRIPTION
**Pull request by leralyshls**
Infinite scroll, format date

### CODE REVIEW

File: dispatcher-client/src/components/dashboard/components/newsWidget/NewsWidget.tsx [hunk 1]
```diff
@@ -1,22 +1,61 @@
...
-const NewsWidget: React.FC = () => {
+const NewsWidget = () => {
+  const [page, setPage] = useState<number>(1);
+  const [news, setNews] = useState<APITypes.Article[]>(
+    topHeadlinesResponces[page].articles
+  );
+  const [hasMore, setHasMore] = useState<boolean>(
+    topHeadlinesResponces[page].articles.length > 0
+  );
+  const observer = useRef<IntersectionObserver>();
```
Comment (avivahls) [inline dispatcher-client/src/components/dashboard/components/newsWidget/NewsWidget.tsx:15]: This is very nice implementation. 
You need to think different when you start using the api calls(With Services) and Redux

File: dispatcher-client/src/components/cardPrimary/CardPrimary.tsx [hunk 2]
```diff
@@ -23,32 +25,35 @@ export interface CardProps {
...
+        <CardImgContainer>
+          <ArticleImg isRTL={isRTL} src={urlToImage ? urlToImage : noImage} />
+        </CardImgContainer>
+        <Article isRTL={isRTL}>
+          <ArticleDetailes>{publishedAt}</ArticleDetailes>
+          <ArticleTitle dir={isRTL ? 'rtl' : 'ltr'} isRTL={isRTL}>
+            {title}
+          </ArticleTitle>
+          <ArticleDetailes>{source.name}</ArticleDetailes>
+          <ArticleContent dir={isRTL ? 'rtl' : 'ltr'} isRTL={isRTL} ref={ref}>
```
[the remark this answers is not in the document]
Comment (leralyshls) [reply dispatcher-client/src/components/cardPrimary/CardPrimary.tsx:45]: Refs cannot be attached to function components, that is why i am using React.forwardRef, so the CardPrimary component can get a ref, and then pass it to one of the dom elements of the card, so when this element becomes visible, the infinite scroll is triggered

[the remark this answers is not in the document]
Comment (avivahls) [reply dispatcher-client/src/components/cardPrimary/CardPrimary.tsx:45]: OK.
now i understand.
You cant rely on card component to do the infinite scroll. 

You need to use the scroll options.
see react-infinite-scroll-component.

File: dispatcher-client/src/components/dashboard/components/newsWidget/NewsWidget.tsx (hunk 1 above)
Comment (avivahls) [reply dispatcher-client/src/components/dashboard/components/newsWidget/NewsWidget.tsx:16]: See other options for infinite scroll

**Review by avivahls — APPROVED** [review]

**Review by avivahls — APPROVED** [review]

**Pull request merged**